src/appleseed/foundation/image/filteredtile.cpp

```diff
@@ -163,4 +165,57 @@ void FilteredTile::atomic_add(
     }
 }
 
+float FilteredTile::compute_weighted_pixel_variance(
+    const float*        main,
+    const float*        second)
+{
+    // Get weights.
+    float main_weight = *main++;
+    main_weight = main_weight == 0.0f ? 0.0f : 1.0f / main_weight;
+    float second_weight = *second++;
+    second_weight = second_weight == 0.0f ? 0.0f : 1.0f / second_weight;
+
+    // Get colors and assign weights.
+    Color4f main_color(abs(main[0]), abs(main[1]), abs(main[2]), abs(main[3]));
```

Comment:
Make sure to include `<cmath>` otherwise you may have bad surprises with `std::abs()`!

Reply:
Actually not sure if computing absolute values here makes sense.

Reply:
It happened that some shading values are negative and makes the metric completly wrong

Reply:
Most of the times it was because of the filter that has negative weights